compiler/src/dotty/tools/dotc/transform/Splicing.scala

```diff
@@ -59,12 +60,12 @@ object Splicing:
  *  ```
  *  '{ ...
  *     val x1: T1 = ???
- *     val x2: T2 = ???
- *     {{{ 0 | T3 | x1, x2 |
- *       (x1$: Expr[T1], x2$: Expr[T2]) => // body of this lambda does not contain references to x1 or x2
- *         (q: Quotes) ?=> f('{ g(${x1$}, ${x2$}) })
- *
- *     }}}
+ *     type U <: T2
+ *     val x2: U = ???
+ *     ${ `0`[U](x1, x2): T3 |
+ *       (U$2: Type[U], x1$: Expr[T], x2$: Expr[U]) => // body of this lambda does not contain references to x1 or x2
+ *         (q: Quotes) ?=> f('{ @SplicedType type U$3 = $[ `0`: U | U$1 ]; g[U$3](${x1$}, ${x2$}) })
```

Comment:
The parameter is called `U$2` but I see `U$1` used here, is that intentional? I also still think it'd be helpful to have some explanation of the `$[ ... ]` syntax somewhere